virttest/utils_test/libvirt.py

```diff
@@ -2814,3 +2808,33 @@ def update_polkit_rule(params, pattern, new_value):
         polkit.polkitd.restart()
     except IOError, e:
         logging.error(e)
+
+
+def get_vol_list(pool_name, vol_check=True, timeout=5):
+    """
+    This is a wrapper to get all volumes of a pool, especially for
+    iscsi type pool as the volume may not appear immediately after
+    iscsi target login.
+
+    :param pool_name: Libvirt pool name
+    :param vol_check: Check if volume and volume path exist
+    :param timeout: Timeout in seconds.
+    :return: A dict include volumes' name(key) and path(value).
+    """
+    poolvol = libvirt_storage.PoolVolume(pool_name=pool_name)
+    vols = utils_misc.wait_for(poolvol.list_volumes, timeout,
+                               text='Waitting for volume show up')
+    if not vol_check:
+        return vols
+
+    # Check volume name
+    if not vols:
```

Comment:
It's better to move this check before return vols. so that exception will raise if wait_for timeout, else TypeError will raise if wait_for timeout occur.

Thanks,
Xu